[BUG] Wrong version of mod downloaded

**Describe the bug**
Wrong version of mod is downloaded

**To Reproduce**
Steps to reproduce the behavior:
1. Update mod
2. Realise wrong mod version was downloaded

**Expected behavior**
Should have downloaded the right version for 1.17.1

**Screenshots**
If applicable, add screenshots to help explain your problem.

** Please complete the following information:**
 - OS: Windows 10
 - Minecraft version 1.17.1
 - Version Latest

** Logs **
Latest log:
[16:58:05] [main/INFO]: Loading for game Minecraft 1.17.1
[16:58:06] [ForkJoinPool-1-worker-7/WARN]: The mod "harvest" contains invalid entries in its mod json:
- Unsupported root entry "links" at line 7 column 10
[16:58:06] [ForkJoinPool-1-worker-3/WARN]: Non-Fabric mod JAR at "D:\Games\MultiMC\instances\1.17.1\.minecraft\mods\illuminations-uninstaller.jar", ignoring
[16:58:07] [main/FATAL]: A critical error occurred
net.fabricmc.loader.discovery.ModResolutionException: Could not find required mod: detailab requires {minecraft @ [1.16.x]}
	at net.fabricmc.loader.discovery.ModResolver.findCompatibleSet(ModResolver.java:198) ~[fabric-loader-0.11.6.jar:?]
	at net.fabricmc.loader.discovery.ModResolver.resolve(ModResolver.java:832) ~[fabric-loader-0.11.6.jar:?]
	at net.fabricmc.loader.FabricLoader.setup(FabricLoader.java:195) ~[fabric-loader-0.11.6.jar:?]
	at net.fabricmc.loader.FabricLoader.load(FabricLoader.java:185) [fabric-loader-0.11.6.jar:?]
	at net.fabricmc.loader.launch.knot.Knot.init(Knot.java:132) [fabric-loader-0.11.6.jar:?]
	at net.fabricmc.loader.launch.knot.KnotClient.main(KnotClient.java:28) [fabric-loader-0.11.6.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:567) ~[?:?]
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:210) [NewLaunch.jar:?]
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:245) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.listen(EntryPoint.java:143) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.main(EntryPoint.java:34) [NewLaunch.jar:?]
Caused by: net.fabricmc.loader.util.sat4j.specs.ContradictionException: Creating Empty clause ?
	at net.fabricmc.loader.util.sat4j.minisat.constraints.cnf.Clauses.propagationCheck(Clauses.java:117) ~[fabric-loader-sat4j-2.3.5.4.jar:?]
	at net.fabricmc.loader.util.sat4j.minisat.constraints.cnf.Clauses.sanityCheck(Clauses.java:97) ~[fabric-loader-sat4j-2.3.5.4.jar:?]
	at net.fabricmc.loader.util.sat4j.minisat.constraints.MixedDataStructureDanielWL.createClause(MixedDataStructureDanielWL.java:81) ~[fabric-loader-sat4j-2.3.5.4.jar:?]
	at net.fabricmc.loader.util.sat4j.minisat.core.Solver.addClause(Solver.java:401) ~[fabric-loader-sat4j-2.3.5.4.jar:?]
	at net.fabricmc.loader.discovery.ModResolver.findCompatibleSet(ModResolver.java:195) ~[fabric-loader-0.11.6.jar:?]
	... 13 more

--- a/src/main/kotlin/xyz/deathsgun/modmanager/ModManager.kt
+++ b/src/main/kotlin/xyz/deathsgun/modmanager/ModManager.kt
@@ -51,12 +51,12 @@ class ModManager : ClientModInitializer {
         @JvmStatic
         fun getVersion(): String {
             return FabricLoader.getInstance().allMods.find { it.metadata.id.equals("modmanager") }
-                ?.metadata?.version?.friendlyString ?: "1.1.0+1.17.1-alpha"
+                ?.metadata?.version?.friendlyString ?: "1.1.0+1.17-alpha"
         }
 
         @JvmStatic
         fun getMinecraftVersion(): String {
-            return MinecraftClient.getInstance()?.game?.version?.name ?: "1.17.1"
+            return MinecraftClient.getInstance()?.game?.version?.releaseTarget ?: "1.17"
         }
     }
 

--- a/src/main/kotlin/xyz/deathsgun/modmanager/update/UpdateManager.kt
+++ b/src/main/kotlin/xyz/deathsgun/modmanager/update/UpdateManager.kt
@@ -250,9 +250,17 @@ class UpdateManager {
 
     private fun installVersion(mod: Mod, version: Version, dir: Path, fabricId: String = mod.slug): ModUpdateResult {
         return try {
-            val asset =
-                version.assets.find { (it.filename.endsWith(".jar") || it.primary) && !it.filename.contains("forge") }
+            val assets = version.assets.filter {
+                (it.filename.endsWith(".jar") || it.primary) && !it.filename.contains("forge")
+            }
+            if (assets.isEmpty()) {
+                return ModUpdateResult.Error(TranslatableText("modmanager.error.update.noFabricJar"))
+            }
+            var asset = assets[0]
+            if (assets.size > 1) {
+                asset = assets.find { it.filename.contains(ModManager.getMinecraftVersion(), true) }
                     ?: return ModUpdateResult.Error(TranslatableText("modmanager.error.update.noFabricJar"))
+            }
             val jar = dir.resolve(asset.filename) // Download into same directory where the old jar was
             val request = HttpRequest.newBuilder(URI.create(asset.url)).GET()
                 .setHeader("User-Agent", "ModManager ${ModManager.getVersion()}").build()

